tfjs-backend-webgpu/src/kernels/select_webgpu.ts

```diff
@@ -25,46 +25,52 @@ export class SelectProgram implements WebGPUProgram {
   variableNames = ['c', 'a', 'b'];
   outputShape: number[];
   shaderKey: string;
-  userCode: string;
   dispatchLayout: {x: number[]};
   dispatch: [number, number, number];
   workPerThread = 4;
   workGroupSize: [number, number, number] = [16, 1, 1];
+  cRank: number;
+  rank: number;
 
   constructor(cRank: number, shape: number[], rank: number) {
     this.outputShape = shape;
-    const size = util.sizeFromShape(this.outputShape);
     this.dispatchLayout = flatDispatchLayout(this.outputShape);
     this.dispatch = computeDispatch(
         this.dispatchLayout, this.outputShape, this.workGroupSize,
         [this.workPerThread, 1, 1]);
 
+    this.cRank = cRank;
+    this.rank = rank;
+    this.shaderKey = 'select';
```

Comment:
missing info from shader key?

Reply:
The `inputs` and `output` shapes are already part of the whole shader key. You can find it in `makeShaderKey` in `webgpu_program.ts`. `cRank` and `rank` are calculated from `inputs` and `output`. So we can skip them here.

Reply:
Got it, sorry I missed this since I didn't review the previous patch. SGTM.